File: backend-java/src/main/java/com/example/pricing/entity/Shop.java
package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 店铺实体，描述用户在不同电商平台下的店铺信息。
 */
@Data
@TableName("shop")
public class Shop {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("user_id")
    private Long userId;

    @TableField("shop_name")
    private String shopName;

    @TableField("platform")
    private String platform;

    @TableField("seller_nick")
    private String sellerNick;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/shop.ts
/**
 * 店铺管理接口封装，负责店铺列表、新增、修改和删除请求。
 */

import request from './request'

export interface Shop {
  id: number
  userId: number
  shopName: string
  platform: string
  sellerNick: string
  createdAt: string
  updatedAt: string
}

export interface ShopCreateDTO {
  shopName: string
  platform: string
  sellerNick?: string
}

export interface ShopUpdateDTO {
  shopName: string
  platform: string
  sellerNick?: string
}

export const getShopList = () => request.get<any>('/shops')
export const createShop = (data: ShopCreateDTO) => request.post<any>('/shops', data)
export const updateShop = (id: number, data: ShopUpdateDTO) => request.put<any>(`/shops/${id}`, data)
export const deleteShop = (id: number) => request.delete<any>(`/shops/${id}`)
